test/openssl/test_pkey_rsa.rb

```diff
@@ -68,6 +68,63 @@ def test_private
     assert(!key6.private?)
   end
 
+  def test_public_eql?
+    key1 = OpenSSL::PKey::RSA.new 2048
+    key2 = OpenSSL::PKey::RSA.new 2048
+
+    assert(key1.public_eql?(key1.public_key))
+    assert(key1.public_key.public_eql?(key1))
+    assert(!key1.public_eql?(key2))
```

Comment:
Please use `Fixtures.pkey("rsa-1")` (or rsa-2/3) for pkey instances, since generating a big RSA key can take some time and slows down the test suite.

It's preferred to do an explicit test a return value: `assert_equal <true|false>, <expr>` over plain `assert`. It will produce a better message if the test fails.

Reply:
Gotcha, will do